Remove pytest traceback when `qa-ctl` is not in `DEBUG` mode

When running a test with `qa-ctl` the results are displayed both on screen and in the file. When the result is displayed on screen, the pytest results are displayed together with the whole traceback, being quite annoying in case some test fails (see example below).

<details>
<summary>Test result</summary>

```
2021-10-28 08:23:55,523 - INFO - Validating input parameters
2021-10-28 08:23:57,603 - INFO - Input parameters validation has passed successfully
2021-10-28 08:23:58,286 - INFO - Starting 2 instances deployment
2021-10-28 08:27:03,924 - INFO - The instances deployment has finished sucessfully
2021-10-28 08:27:03,926 - INFO - Checking hosts SSH connection
2021-10-28 08:28:22,903 - INFO - Hosts connection OK. The instances are accessible via ssh
2021-10-28 08:28:22,903 - INFO - Provisioning 2 instances
2021-10-28 08:28:57,767 - INFO - Performing a Wazuh installation healthcheck in 10.150.50.2 host
2021-10-28 08:29:20,074 - INFO - Performing a Wazuh installation healthcheck in 10.150.50.3 host
2021-10-28 08:29:37,218 - INFO - Provisioning the 10.150.50.2 host with the Wazuh QA framework using 2020-qa-ctl-windows branch.
2021-10-28 08:29:54,702 - INFO - Provisioning the 10.150.50.3 host with the Wazuh QA framework using 2020-qa-ctl-windows branch.
2021-10-28 08:30:55,176 - INFO - The instances have been provisioned sucessfully
2021-10-28 08:30:55,177 - INFO - Launching 1 tests
2021-10-28 08:30:55,177 - INFO - Waiting for tests to finish
2021-10-28 08:31:09,208 - INFO - Running C:\Users\vagrant\AppData\Local\Temp\wazuh_qa_ctl\wazuh-qa\tests\integration\test_fim\test_registry\test_registry_restrict\test_registry_restrict.py test on ['10.150.50.2'] hosts
2021-10-28 08:32:32,297 - INFO - 

============================= test session starts =============================
platform win32 -- Python 3.9.7, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: C:\Users\vagrant\AppData\Local\Temp\wazuh_qa_ctl\wazuh-qa\tests\integration, configfile: pytest.ini
plugins: html-2.0.1, metadata-1.11.0, testinfra-5.0.0
collected 24 items

test_fim\test_registry\test_registry_restrict\test_registry_restrict.py F [  4%]
.....ssssssssssss......                                                  [100%]

================================== FAILURES ===================================
_ test_restrict_value[get_configuration0-HKEY_LOCAL_MACHINE-SOFTWARE\\Classes\\testkey-256-value_restrict-True-tags_to_apply0] _

key = 'HKEY_LOCAL_MACHINE', subkey = 'SOFTWARE\\Classes\\testkey', arch = 256
value_name = 'value_restrict', triggers_event = True
tags_to_apply = {'value_restrict'}
get_configuration = {'apply_to_modules': ['test_registry_restrict'], 'metadata': {'fim_mode': 'scheduled', 'restrict_key': 'key_restrict$'...utes': [...], 'value': 'HKEY_LOCAL_MACHINE\\SOFTWARE\\testkey'}}], 'section': 'syscheck'}], 'tags': ['value_restrict']}
configure_environment = None, restart_syscheckd = None
wait_for_fim_start = None

    @pytest.mark.parametrize('key, subkey, arch, value_name, triggers_event, tags_to_apply', [
        (key, sub_key_1, KEY_WOW64_64KEY, valid_value_name, True, {'value_restrict'}),
        (key, sub_key_2, KEY_WOW64_32KEY, valid_value_name, True, {'value_restrict'}),
        (key, sub_key_2, KEY_WOW64_64KEY, valid_value_name, True, {'value_restrict'}),
        (key, sub_key_1, KEY_WOW64_64KEY, no_valid_value_name, False, {'value_restrict'}),
        (key, sub_key_2, KEY_WOW64_32KEY, no_valid_value_name, False, {'value_restrict'}),
        (key, sub_key_2, KEY_WOW64_64KEY, no_valid_value_name, False, {'value_restrict'})
    ])
    def test_restrict_value(key, subkey, arch, value_name, triggers_event, tags_to_apply,
                            get_configuration, configure_environment, restart_syscheckd,
                            wait_for_fim_start):
        '''
        description: Check if the 'wazuh-syscheckd' daemon detects or ignores events in monitored registry entries
                     depending on the value set in the 'restrict_value' attribute. This attribute limit checks to
                     values that match the entered string or regex and its name. For this purpose, the test will
                     monitor a key, create testing values inside it, and make operations on that values. Finally,
                     the test will verify that FIM 'added' and 'modified' events are generated only for the testing
                     values that are not restricted.
    
        wazuh_min_version: 4.2.0
    
        parameters:
            - key:
                type: str
                brief: Path of the registry root key (HKEY_* constants).
            - subkey:
                type: str
                brief: The registry key being monitored by syscheck.
            - arch:
                type: str
                brief: Architecture of the registry.
            - value_name:
                type: str
                brief: Name of the testing value that will be created.
            - triggers_event:
                type: bool
                brief: True if an event must be generated, False otherwise.
            - tags_to_apply:
                type: set
                brief: Run test if matches with a configuration identifier, skip otherwise.
            - get_configuration:
                type: fixture
                brief: Get configurations from the module.
            - configure_environment:
                type: fixture
                brief: Configure a custom environment for testing.
            - restart_syscheckd:
                type: fixture
                brief: Clear the 'ossec.log' file and start a new monitor.
            - wait_for_fim_start:
                type: fixture
                brief: Wait for realtime start, whodata start, or end of initial FIM scan.
    
        assertions:
            - Verify that FIM events are only generated for operations in monitored values
              that do not match the 'restrict_value' attribute.
            - Verify that FIM 'ignoring' events are generated for monitored values that are restricted.
    
        input_description: A test case (value_restrict) is contained in external YAML file
                           (wazuh_restrict_conf.yaml) which includes configuration settings
                           for the 'wazuh-syscheckd' daemon. That is combined with the testing
                           registry keys to be monitored defined in this module.
    
        expected_output:
            - r'.*Sending FIM event: (.+)$' ('added', 'modified' events)
            - r'.*Ignoring entry .* due to restriction .*'
    
        tags:
            - scheduled
            - time_travel
        '''
        check_apply_test(tags_to_apply, get_configuration['tags'])
        # This shouldn't create an alert because the key is already created
        key_h = create_registry(registry_parser[key], subkey, arch)
        # Create values
        modify_registry_value(key_h, value_name, REG_SZ, "added")
        # Go ahead in time to let syscheck perform a new scan
        check_time_travel(get_configuration['metadata']['fim_mode'] == 'scheduled', monitor=wazuh_log_monitor)
    
>       event = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
                                        callback=callback_detect_event, accum_results=2 if triggers_event else 1).result()

test_fim\test_registry\test_registry_restrict\test_registry_restrict.py:186: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\..\..\..\Programs\Python\Python39\lib\site-packages\wazuh_testing-4.3.0-py3.9.egg\wazuh_testing\tools\monitoring.py:195: in start
    self._result = monitor.start(timeout=timeout, callback=callback, accum_results=accum_results,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <wazuh_testing.tools.monitoring.QueueMonitor object at 0x000001A3790B75B0>
timeout = 10, callback = <function callback_detect_event at 0x000001A378FB71F0>
accum_results = 2, update_position = True, timeout_extra = 0, error_message = ''

    def start(self, timeout=-1, callback=_callback_default, accum_results=1, update_position=True, timeout_extra=0,
              error_message=''):
        """Start the queue monitoring until the stop method is called."""
        if not self._continue:
            self._continue = True
            self._abort = False
            result = None
    
            while self._continue:
                if self._abort:
                    self.stop()
                    if error_message:
                        logger.error(error_message)
                        logger.error(f"Results accumulated: "
                                     f"{len(result) if isinstance(result, list) else 0}")
                        logger.error(f"Results expected: {accum_results}")
>                   raise TimeoutError(error_message)
E                   TimeoutError

..\..\..\..\..\Programs\Python\Python39\lib\site-packages\wazuh_testing-4.3.0-py3.9.egg\wazuh_testing\tools\monitoring.py:457: TimeoutError
---------------------------- Captured stdout setup ----------------------------


The Wazuh service was stopped successfully.





The Wazuh service was started successfully.



---------------------------- Captured stderr setup ----------------------------
2021-10-27 23:31:18,225 - wazuh_testing - INFO - Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\Classes\testkey
2021-10-27 23:31:18,225 - wazuh_testing - INFO - Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\Classes\testkey
2021-10-27 23:31:18,225 - wazuh_testing - INFO - Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\testkey
2021-10-27 23:31:18,225 - wazuh_testing - INFO - Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\testkey
----------------------------- Captured log setup ------------------------------
INFO     wazuh_testing:fim.py:443 Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\Classes\testkey
INFO     wazuh_testing:fim.py:443 Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\Classes\testkey
INFO     wazuh_testing:fim.py:443 Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\testkey
INFO     wazuh_testing:fim.py:443 Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\testkey
---------------------------- Captured stdout call -----------------------------


Thursday, October 28, 2021 12:31:18 PM





---------------------------- Captured stderr call -----------------------------
2021-10-27 23:31:18,881 - wazuh_testing - INFO - Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\Classes\testkey
2021-10-27 23:31:18,881 - wazuh_testing - INFO - Modifying value 'value_restrict' of type REG_SZ and value 'added'
2021-10-28 12:31:18,749 - wazuh_testing - INFO - Changing the system clock from 2021-10-27 23:31:18.881702 to 2021-10-28 12:31:18.749138
------------------------------ Captured log call ------------------------------
INFO     wazuh_testing:fim.py:443 Creating registry key HKEY_LOCAL_MACHINE\SOFTWARE\Classes\testkey
INFO     wazuh_testing:fim.py:619 Modifying value 'value_restrict' of type REG_SZ and value 'added'
INFO     wazuh_testing:fim.py:1374 Changing the system clock from 2021-10-27 23:31:18.881702 to 2021-10-28 12:31:18.749138
- generated html file: file://C:\Users\vagrant\AppData\Local\Temp\wazuh_qa_ctl\wazuh-qa\tests\integration\reports\test_report_2021_10_28_08_31_09_208301.html -
=========================== short test summary info ===========================
FAILED test_fim/test_registry/test_registry_restrict/test_registry_restrict.py::test_restrict_value[get_configuration0-HKEY_LOCAL_MACHINE-SOFTWARE\\Classes\\testkey-256-value_restrict-True-tags_to_apply0]
================== 1 failed, 11 passed, 12 skipped in 35.98s ==================



2021-10-28 08:32:32,298 - INFO - The test run is finished
2021-10-28 08:32:32,299 - INFO - The results of C:\Users\vagrant\AppData\Local\Temp\wazuh_qa_ctl\wazuh-qa\tests\integration\test_fim\test_registry\test_registry_restrict\test_registry_restrict.py tests have been saved in /tmp/wazuh_qa_ctl/test_registry_restrict_1635402238.273011
2021-10-28 08:32:32,299 - INFO - Destroying 2 instances
2021-10-28 08:32:39,081 - INFO - The instances have been destroyed sucessfully

```

</details>

It is requested to remove only traceback from the screen output when `qa-ctl` is not in DEBUG mode, keeping the pytest results stored in the file with the complete information. For this purpose it is proposed to trim the output shown only the p test session starts`. 

For example:

```
============================= test session starts =============================
platform win32 -- Python 3.9.7, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: C:\Users\vagrant\AppData\Local\Temp\wazuh_qa_ctl\wazuh-qa\tests\integration, configfile: pytest.ini
plugins: html-2.0.1, metadata-1.11.0, testinfra-5.0.0
collected 24 items

test_fim\test_registry\test_registry_restrict\test_registry_restrict.py F [  4%]
.....ssssssssssss......                                                  [100%]

```

--- a/deps/wazuh_testing/wazuh_testing/qa_ctl/run_tests/pytest.py
+++ b/deps/wazuh_testing/wazuh_testing/qa_ctl/run_tests/pytest.py
@@ -1,5 +1,5 @@
 import os
-
+import re
 from datetime import datetime
 from tempfile import gettempdir
 from wazuh_testing.qa_ctl.run_tests.test_result import TestResult
@@ -194,8 +194,28 @@ class Pytest(Test):
                                  plain_report_file_path=os.path.join(self.tests_result_path, plain_report_file_name),
                                  test_name=self.tests_path)
 
+        # Trim the result report for a more simple and readable output
+        output_result = str(self.result)
+        error_fail_pattern =  re.compile('^=*.(ERRORS|FAILURES|short test summary info).*=$', re.M)
+        test_summary_pattern = re.compile('^=*.(short test summary info).*=$', re.M)
+
+        error_case = re.search(error_fail_pattern, output_result)
+        if error_case is not None:
+            test_summary_case = re.search(test_summary_pattern, output_result)
+            if test_summary_case is not None:
+                test_result_message = test_summary_case.group(0)
+                result_output = output_result[output_result.index(test_result_message):]
+            
+            error_case_message = error_case.group(0)
+            trimmed_output = output_result[:output_result.index(error_case_message)]
+            trimmed_output += result_output
         # Print test result in stdout
-        if self.qa_ctl_configuration.logging_enable:
-            Pytest.LOGGER.info(self.result)
+            if self.qa_ctl_configuration.logging_enable:
+                Pytest.LOGGER.info(trimmed_output)
+            else:
+                print(trimmed_output)
         else:
-            print(self.result)
+            if self.qa_ctl_configuration.logging_enable:
+                Pytest.LOGGER.info(self.result)
+            else:
+                print(self.result)

--- a/deps/wazuh_testing/wazuh_testing/qa_ctl/run_tests/pytest.py
+++ b/deps/wazuh_testing/wazuh_testing/qa_ctl/run_tests/pytest.py
@@ -86,6 +86,36 @@ class Pytest(Test):
 
         super().__init__(tests_path, tests_run_dir, tests_result_path, modules, component, system)
 
+    def __output_trimmer(self, result):
+        """This function trims the obtained results in order to get a more readable output information
+            when executing qa-ctl
+
+        Args:
+            result (Test resutl object): object containing all the results obtained from the test
+
+        Return:
+            output_result (string): String containing the trimmed output 
+        """
+        output_result = str(result)
+        error_fail_pattern =  re.compile('^=*.(ERRORS|FAILURES).*=$', re.M)
+        test_summary_pattern = re.compile('^=*.(short test summary info).*=$', re.M)
+
+        # Check for any error or failure message case in the test result output
+        error_case = re.search(error_fail_pattern, output_result)
+        if error_case is not None:
+            # Checks if there is any test summary info at the end of the result output
+            test_summary_case = re.search(test_summary_pattern, output_result)
+            test_summary_output = ''
+            if test_summary_case is not None:
+                test_result_message = test_summary_case.group(0)
+                test_summary_output = output_result[output_result.index(test_result_message):]
+            
+            error_case_message = error_case.group(0)
+            output_result = output_result[:output_result.index(error_case_message)]
+            output_result += test_summary_output
+
+        return output_result
+
     def run(self, ansible_inventory_path):
         """Executes the current test with the specified options defined in attributes and bring back the reports
             to the host machine.
@@ -244,24 +274,11 @@ class Pytest(Test):
                                  test_name=self.tests_path)
 
         # Trim the result report for a more simple and readable output
-        output_result = str(self.result)
-        error_fail_pattern =  re.compile('^=*.(ERRORS|FAILURES|short test summary info).*=$', re.M)
-        test_summary_pattern = re.compile('^=*.(short test summary info).*=$', re.M)
-
-        # Checks for any error or failure message case in the test result output
-        error_case = re.search(error_fail_pattern, output_result)
-        if error_case is not None:
-            # Checks if there is any test summary info at the end of the result output
-            test_summary_case = re.search(test_summary_pattern, output_result)
-            test_summary_output = ""
-            if test_summary_case is not None:
-                test_result_message = test_summary_case.group(0)
-                test_summary_output = output_result[output_result.index(test_result_message):]
-            
-            error_case_message = error_case.group(0)
-            output_result = output_result[:output_result.index(error_case_message)]
-            output_result += test_summary_output
-
+        if Pytest.LOGGER.level != 10: 
+            output_result = self.__output_trimmer(self.result)
+        else:
+            output_result = str(self.result)
+        
         # Print test result in stdout
         if self.qa_ctl_configuration.logging_enable:
             if os.path.exists(self.result.plain_report_file_path):

